arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -141,109 +143,136 @@ def start_pcp(
                 "The pmlogger output file was not found; Unable to process results."
             )
 
-        pcp2_flags = [
-            "-a",
-            "pmlogger-out",
-            "-t",
-            str(params.pmlogger_interval),
-            "-f",
-            # pmrep doesn't accept %z for the timezone,
-            # so we'll get it explicitly via datetime
-            f"%FT%T.%f{str(datetime.now().astimezone())[-6:]}",
-            "-c",
-            pmrep_conf_path,
-            "-P",
-            "6",
-        ]
-
-        # Initialized pcp2json command
-        pcp2json_cmd = [
-            "/usr/bin/pcp2json",
-            "-E",
-        ]
-
-        pcp2json_cmd.extend(pcp2_flags)
+        post_process_params = {
+            "pmlogger_metrics": params.pmlogger_metrics,
+            "pmlogger_interval": params.pmlogger_interval,
+            "pmrep_conf_path": params.pmrep_conf_path,
+            "generate_csv": params.generate_csv,
+            "flatten": params.flatten,
+            "archive_path": ".",
+        }
+
+        return post_process(
+            params=post_process_params_schema.unserialize(post_process_params),
+            run_id="post-process",
+        )
 
-        # The list of metrics to collect is appended to the pcp2json command
-        pcp2json_cmd.extend(metrics)
 
-        print(f"Reporting metrics for: {params.pmlogger_metrics}")
+@plugin.step(
+    id="post-process",
+    name="Post-Process PCP Archive",
+    description="Processes an existing PCP archive into a machine-readable format",
+    outputs={"success": PerfOutput, "error": Error},
+)
+def post_process(
+    params: PostProcessParams,
+) -> typing.Tuple[str, typing.Union[PerfOutput, Error]]:
+
+    # Parse metrics from input
+    metrics = params.pmlogger_metrics.split()
+
+    pcp2_flags = [
+        "-a",
+        params.archive_path,
+        "-t",
+        str(params.pmlogger_interval),
+        "-f",
+        # pmrep doesn't accept %z for the timezone,
+        # so we'll get it explicitly via datetime
+        f"%FT%T.%f{str(datetime.now().astimezone())[-6:]}",
+        "-c",
+        params.pmrep_conf_path,
+        "-P",
+        "6",
+    ]
+
+    # Initialized pcp2json command
+    pcp2json_cmd = [
+        "/usr/bin/pcp2json",
+        "-E",
+    ]
+
+    pcp2json_cmd.extend(pcp2_flags)
+
+    # The list of metrics to collect is appended to the pcp2json command
+    pcp2json_cmd.extend(metrics)
```

Comment:
This code comment doesn't say anything which the code doesn't already express.

Ditto for the one at line 203.

Reply:
I hope in essence that is true of ALL code comments. :laughing:

Reply:
A lot hangs on how expressive the code is, on the choices of the names of functions and variables.  Ideally, the code should read enough like English that code comments are superfluous, particularly within the domain of a single line.

Obviously, that's not always true.  There are obscure or arcane operators and language constructs which are compact, powerful, worthwhile, but not expressive.  Use of those might warrant explanation in-line.

And, there are situations where it makes good sense to provide a code comment to lay out an overview of a block or function, in which the text describes the algorithm, approach, or technique which it employs, illuminating details would not necessarily be clear from a looking at a single statement or sequence in the code.

But, here, we have a piece of code which is modifying the existing definition of the `pcp2json` `c`o`m`man`d` by `extend`ing (i.e., _appending_) the list of `metrics` to it...the comment is literally just restating what the line of code is already expressing in nearly identical terms.  So, it's just needlessly taking up space.